File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/domain/TourBookingStatus.java
package com.dominikgaller.alpinebooking.booking.core.domain;

/**
 * Lifecycle states of a {@link TourBooking} aggregate.
 *
 * <p>Allowed transitions:
 * <ul>
 *   <li>REQUESTED → CONFIRMED</li>
 *   <li>REQUESTED → CANCELLED</li>
 *   <li>CONFIRMED → CANCELLED</li>
 *   <li>CONFIRMED → ACTIVE</li>
 *   <li>ACTIVE → COMPLETED</li>
 * </ul>
 *
 * <p>SDD: See {@code documentation/domain/aggregate-tour-booking.spec.md}, section 3.
 */
public enum TourBookingStatus {
    REQUESTED,
    CONFIRMED,
    CANCELLED,
    ACTIVE,
    COMPLETED
}

File: src/main/kotlin/com/example/contractmanagement/booking/core/domain/tourbooking/TourBookingStatus.kt
package com.example.contractmanagement.booking.core.domain.tourbooking

/**
 * Lifecycle of a [TourBooking].
 *
 * `CANCELLED` and `COMPLETED` are terminal. The transitions that reach each state are
 * defined by the aggregate, not here — an enum lists states, it does not own the rules.
 *
 * SDD: see `documentation/domain/aggregate-tour-booking.spec.md`.
 */
enum class TourBookingStatus {
    REQUESTED,
    CONFIRMED,
    CANCELLED,
    ACTIVE,
    COMPLETED,
}
